### PR TITLE
feat: expiry for cache keys [CW-3038]

### PR DESCRIPTION
**Pull request by scmmishra**
This PR tackles to problems with current implementation

1. In a rare case, the cache might show stale data, this means unless something changes for that model, or someone resets the cache keys manually, it will continue to show stale data
2. The value `0000000000` also is a valid key, this means when someone reset the cache from superadmin, it won't reset the cache for someone holding the cache key `0000000000`
3. Destroy event didn't update cache

This PR addresses this with the following changes
1. Set a default expiry of 72 hours for the cache keys, this can be changed to a higher number, this ensures that new data is fetched after every 3 days
2. When reseting the cache, we set the value to the latest timestamp, that way the conflict for the `zero` epoch is solved
3. Trigger `update_account_cache` on `destroy`

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: app/models/concerns/cache_keys.rb [hunk 1]
```diff
@@ -18,26 +20,26 @@ def cache_keys
...
-      invalidate_cache_key_for(model.name.underscore)
+      udpate_cache_key_for_account(id, model.name.underscore)
     end
+
+    dispatch_cache_update_event
   end
 
   private
 
+  def udpate_cache_key_for_account(account_id, key)
```
Comment (sojan-official) [inline app/models/concerns/cache_keys.rb:38]: `update_` ?

**Comment by pranavrajs** [comment]
@scmmishra The specs seems to be broken. Can you please take a look?

Screenshot 2024-01-26 at 11 13 12 AM

**Review by sojan-official — COMMENTED** [review]
LGTM mostly. just a minor comment on the spec

File: spec/models/inbox_spec.rb [hunk 2]
```diff
@@ -250,5 +250,15 @@
...
     end
+
+    it 'updates the cache key after update' do
+      expect(inbox.account).to receive(:update_cache_key).with('inbox')
+      inbox.update(name: 'New Name')
+    end
+
+    it 'updates the cache key after touch' do
+      expect(inbox.account).to receive(:update_cache_key).with('inbox')
+      inbox.touch # rubocop:disable Rails/SkipsModelValidations
```
[the remark this answers is not in the document]
Comment (scmmishra) [reply spec/models/inbox_spec.rb:261]: It's already handled in the same file

```
    it 'resets cache key if there is an update in the channel' do
      channel = inbox.channel
      channel.update(widget_color: '#fff')

      expect(Rails.configuration.dispatcher).to have_received(:dispatch)
        .with(
          'account.cache_invalidated',
          kind_of(Time),
          account: inbox.account,
          cache_keys: inbox.account.cache_keys
        )
    end
```

[the remark this answers is not in the document]
Comment (sojan-official) [reply spec/models/inbox_spec.rb:261]: Awesome. Thanks for linking this.

**Review by sojan-official — APPROVED** [review]

**Pull request merged**